### PR TITLE
feat: handle error messages for merge action

### PR DESCRIPTION
**Pull request by AnaghHegde**
## Description

> Group the result of merge command into one standard

Fixes # (issue)

> if no issue exists, please create an issue and ask the maintainers about this first

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

- Locally
- Postman

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/controllers/GitController.java [hunk 1]
```diff
@@ -154,17 +155,16 @@ public GitController(GitService service) {
...
                 .map(result -> new ResponseDTO<>(HttpStatus.OK.value(), result, null));
     }
 
     @GetMapping("/merge/status/{defaultApplicationId}")
-    public Mono<ResponseDTO<MergeStatus>> mergeStatus(@PathVariable String defaultApplicationId,
-                                                      @RequestParam String sourceBranch,
-                                                      @RequestParam String destinationBranch) {
+    public Mono<ResponseDTO<MergeStatusDTO>> mergeStatus(@PathVariable String defaultApplicationId,
+                                                         @RequestParam String sourceBranch,
+                                                         @RequestParam String destinationBranch) {
```
Comment (abhvsn) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/controllers/GitController.java:167]: Are we still using branchNames for this API from parth params?

**Review by abhvsn — COMMENTED** [review]
LGTM just a minor comment

File: app/server/appsmith-server/src/main/java/com/appsmith/server/controllers/GitController.java (hunk 1 above)
Comment (AnaghHegde) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/controllers/GitController.java:167]: My bad, i will remove this.

**Review by abhvsn — DISMISSED** [review]

**Review by abhvsn — DISMISSED** [review]

**Review by abhvsn — APPROVED** [review]

**Review by abhvsn — APPROVED** [review]

**Pull request merged**